examples/timeseries/eeg_bci_ssvepformer.py

```diff
@@ -0,0 +1,655 @@
+        optimizer=sgd,
+        metrics=["accuracy"],
+        jit_compile=True,
+    )
+
+    history = model.fit(
+        train_dataset,
+        batch_size=BATCH_SIZE,
+        epochs=EPOCHS,
+        validation_data=test_dataset,
+        verbose=0,
+    )
+    loss, acc = model.evaluate(test_dataset)
+    return acc * 100
+
+
+"""
+## Run evaluation
+"""
+
+channels = X.shape[2]
+samples = X.shape[1]
+input_shape = (channels, samples)
+n_classes = 12
+
+model = create_ssvepformer(
+    input_shape, fs, resolution, band, channels, n_classes, DROP_RATE
+)
+model.summary()
+
+"""
+## Evaluation on all subjects following a leave-one-subject out data repartition scheme
+"""
+
+accs = np.zeros(10)
+
+for subject in range(10):
+    print(f"Testing subject: {subject+ 1}")
+
+    # create train / test folds
+    folds = np.delete(np.arange(10), subject)
+    train_index = folds
+    test_index = [subject]
+
+    # create data split for each subject
+    x_train, y_train = concatenate_subjects(X, Y, train_index)
+    x_val, y_val = concatenate_subjects(X, Y, test_index)
+
+    # train and evaluate a fold and compute the time it takes
+    acc = evaluate_subject(x_train, y_train, x_val, y_val, input_shape)
+
+    accs[subject] = acc
+
+print(f"\nAccuracy Across Subjects: {accs.mean()} % std: {np.std(accs)}")
+
+"""
+and that's it! we see how some subjects with no data on the training set still can achieve
+almost a 100% correct commands and others show poor performance around 50%. In the original
+paper using PyTorch the average accuracy was 84.04% with 17.37 std. we reached the same
+values knowing the stochastic nature of deep learning.
+"""
+
+"""
+# Visualizations
+
+Training and inference times comparison between the different backends (Jax, Tensorflow
+and PyTorch) on the three GPUs available with Colab Free/Pro/Pro+: T4, L4, A100.
+
+
+"""
+
+"""
+## Training Time
+
+![training_time](/img/eeg_bci_ssvepformer/eeg_ssvepformer_keras_training_time.png)
+"""
+
+"""
+# Inference Time
+
+![inference_time](/img/eeg_bci_ssvepformer/eeg_ssvepformer_keras_inference_time.png)
+"""
+
+"""
+the Jax backend was the best on training and inference in all the GPUs, the PyTorch was
+exremely slow due to the jit compilation option being disable because of the complex
+data type calculated by FFT which is not supported by the PyTorch jit compiler.
+"""
+
+"""
+# Acknowledgment
+
+I thank Chris Perry [X](https://x.com/thechrisperry) @GoogleColab for supporting this
+work with GPU compute.
+"""
+
+"""
+# References
+[1] Chen, J. et al. (2023) ‘A transformer-based deep neural network model for SSVEP
+classification’, Neural Networks, 164, pp. 521–534. Available at: https://doi.org/10.1016/j.neunet.2023.04.045.
+
+[2] Nakanishi, M. et al. (2015) ‘A Comparison Study of Canonical Correlation Analysis
+Based Methods for Detecting Steady-State Visual Evoked Potentials’, Plos One, 10(10), p.
+e0140703. Available at: https://doi.org/10.1371/journal.pone.0140703
+"""
```

Comment:
Can you add a newline at the end here?

It's failing black right now: https://github.com/keras-team/keras-io/actions/runs/13016143246/job/36753267674?pr=2025